### DISCUSSION TITLE
Cache invalidation explain

### DISCUSSION THREAD

**Discussion by aluzzardi** [dagger/dagger#12056] [Ideas]
# Cache invalidation explain

It's very common for an action (or a set of actions) to be re-executed given that "something" in the build cache changed. While speaking with @gerhard today across several topics, we came up with the idea that it'd be great if Dagger could output some optional detailed information about *why* the cache is being invalidated. I think this could potentially help to prevent mistakes and/or find issues/improvements in already executed pipelines. 

As  side nice-to-have it'd be also great to be able to do something like `dagger do test --explain` or something similar to ask Dagger if the next run will hit some cache invalidation and `why` without actually running the action.

_Originally posted by @marcosnils in https://github.com/dagger/dagger/discussions/2052#discussioncomment-2533382_

**Comment by shykes** [comment]
This would be a really cool feature for Cloud debugging... :) cc @marcosnils

**Comment by shykes** [comment]
@vito found this old issue, thought you'd find it interesting in the context of your prototyping work :)

**Comment by marcosnils** [comment]
> This would be a really cool feature for Cloud debugging... :) cc @marcosnils

Indeed, I was the one originally posting the question ;)
